spec/unit/provider/gluster_peer_spec.rb

```diff
@@ -1,15 +1,14 @@
 require 'spec_helper'
-require 'unit/provider/helpers'
-require 'unit/provider/fake_gluster'
+require 'unit/helpers'
 
-describe Puppet::Type.type(:gluster_peer).provider(:gluster_peer) do
+peer_type = Puppet::Type.type(:gluster_peer)
+
+describe peer_type.provider(:gluster_peer), :unit => true do
   on_supported_os.each do |os, facts|
     context "on #{os}" do
       before :each do
-        Facter.clear
-        facts.each do |k, v|
-          Facter.stubs(:fact).with(k).returns Facter.add(k) { setcode { v } }
-        end
+        allow(Facter).to receive(:collection, &make_fake_collection(facts))
+        Facter.reset
```

Comment:
The `allow` method is in scope because it's part of the rspec mocking stuff, so I can't just wrap the first line in a helper method somewhere else. (I'm sure there's a way to do it, I just haven't found it yet.)